router/src/validation.rs

```diff
@@ -964,4 +961,76 @@ mod tests {
 
         assert_eq!(valid_request.top_n_tokens, 0);
     }
+
+    static PIXEL_GIF: &str = "R0lGODdhAQABAIEAAP///wAAAAAAAAAAACwAAAAAAQABAAAIBAABBAQAOw==";
+
+    #[tokio::test]
+    async fn test_prepare_input_chunks() {
+        let pixel_data = STANDARD.decode(PIXEL_GIF).unwrap();
+
+        let tokenizer = Some(get_tokenizer().await);
+
+        let max_best_of = 2;
+        let max_stop_sequence = 3;
+        let max_top_n_tokens = 4;
+        let max_input_length = 5;
+        let max_total_tokens = 6;
+        let disable_grammar_support = true;
+        let workers = 1;
+        let configs = &[
+            Config::Idefics,
+            Config::Idefics2(Idefics2 {}),
+            Config::LlavaNext(LlavaNext {
+                text_config: TextConfig {},
+                vision_config: VisionConfig {
+                    image_size: 1,
+                    patch_size: 1,
+                },
+                image_grid_pinpoints: vec![],
+            }),
+            Config::Paligemma(Paligemma {
+                text_config: PaliTextConfig {
+                    num_image_tokens: 1,
+                },
+            }),
+        ];
```

Comment:
I'm never a big fan of parameterized tests.

The forloops tend to obscure the logic (and for me a test is only valuable if it's extremely readable and that the reader can infer quite simply if the test is wrong and needs an update, or if the business logic is wrong and bug needs to be fixed).

Here for instance I don't think we're actually testing anything different, since they all endup in the same codepath no ?

Reply:
Simplifying this now to just test one config, I have prepared a next PR that merges the code paths (but needs this one).

Reply:
The test is not parametrized anymore.